lm_eval/tasks/__init__.py

```diff
@@ -1,230 +1,312 @@
-
-        task_config = utils.load_yaml_config(yaml_path, task_config)
-        var_configs = check_prompt_config(
-            {
-                **base_config,
-                **task_config,
-                **{"group": group},
-                **task_name_config,
-            },
-            yaml_path=os.path.dirname(yaml_path),
-        )
-        for config in var_configs:
-            register_configurable_task(config)
-
-    task_names = utils.pattern_match(task_list, ALL_TASKS)
-    for task in task_names:
-        if (task in TASK_REGISTRY) or (task in GROUP_REGISTRY):
-            if group in GROUP_REGISTRY:
-                GROUP_REGISTRY[group].append(task)
-            else:
-                GROUP_REGISTRY[group] = [task]
-                ALL_TASKS.add(group)
-
-    return 0
-
-
-def check_prompt_config(
-    config: Dict[str, str], yaml_path: str = None
-) -> List[Dict[str, str]]:
-    all_configs = []
-    if "use_prompt" in config:
-        prompt_list = prompts.load_prompt_list(
-            use_prompt=config["use_prompt"],
-            dataset_name=config["dataset_path"],
-            subset_name=config["dataset_name"] if "dataset_name" in config else None,
-            yaml_path=yaml_path,
-        )
-        for idx, prompt_variation in enumerate(prompt_list):
-            all_configs.append(
-                {
-                    **config,
-                    **{"use_prompt": prompt_variation},
-                    **{
-                        "task": "_".join(
-                            [
-                                config["task"]
-                                if "task" in config
-                                else get_task_name_from_config(config),
-                                prompt_variation.split("/")[-1]
-                                if ".yaml" in prompt_variation
-                                else prompt_variation,
-                            ]
-                        )
-                    },
-                    **{"output_type": "generate_until"},
-                }
+    def __init__(
+        self,
+        verbosity="INFO",
+        include_path=None
+        ) -> None:
+
+        self.verbosity = verbosity
+        self.include_path = include_path
+        self.logger = utils.eval_logger
+        self.logger.setLevel(getattr(logging, f"{verbosity}"))
+
+        self.ALL_TASKS = self.initialize_tasks(
+            include_path=include_path
             )
-    else:
-        all_configs.append(config)
-    return all_configs
 
+        self.task_group_map = collections.defaultdict(list)
 
-def get_task_name_from_config(task_config: Dict[str, str]) -> str:
-    if "dataset_name" in task_config:
-        return "{dataset_path}_{dataset_name}".format(**task_config)
-    else:
-        return "{dataset_path}".format(**task_config)
+    def initialize_tasks(self, include_path=None):
 
+        all_paths = [os.path.dirname(os.path.abspath(__file__)) + "/"]
```

Comment:
Does this still retain the functionality for someone to not wish to load any of the stock tasks, and only load the ones in `include_path`?

I don't know for sure if people are currently relying on this, but it seems nice to support.

maybe a flag in TaskManager init can control load_default_library or something?

Reply:
The stock tasks will only be indexed and not loaded at all, so if the user has some duplicate tasks from `include_path` it would work.